Update UFS model with physics updates for GFSv17

JessicaMeixner-NOAA opened the pull request:
<!--
  *** PLEASE READ ***

  Any PRs not following this template will be closed.

  Please delete all these comments before submitting the PR.

  Please use a short (<60 char), descriptive title for the PR title above. It should complete the sentence "If merged, this PR will _____". Capitalize the first word and do not end with a period.

  No content should appear above the "Description" header.

  If this PR is not merge-ready (e.g. it depends on other PRs not yet merged), please mark it as draft until it is ready.

  PRs should meet these guidelines:
  - Each PR should address ONE topic and have an associated issue.
  - No hard-coded paths or personal directories.
  - No temporary or backup files should be committed (including logs).
  - Any code that you disabled by being commented out should be removed or reenabled.
-->
# Description
<!-- This description will become the commit message for the PR. -->
<!--
  Solely pointing to an issue is not an adequate description!

  Please use this format for your description:

  Describe your changes. Focus on the *what* and *why*. The *how* will be evident from the changes. In particular, be sure to note any interface changes, such as command line syntax, that will need to be communicated to users.

  At the end of your description, please be sure to add the issue this PR solves using the word "Resolves". If there are any issues that are related but not yet resolved (including in other repos), you may use "Refs".

  Resolves #1234
  Refs #4321
  Refs NOAA-EMC/repo#5678
-->
<!-- For more on writing good commit messages, see https://cbea.ms/git-commit/ -->

Updates model with the last of the known physics updates (at this time) for the model.   Changes in the model can be seen here: 
https://github.com/ufs-community/ufs-weather-model/compare/9193010aa52f17a7b3618fd75b4e3c03a79c3c59...1f0ab4322b0c81e48a0db5a202a427d374e29be0

This includes https://github.com/ufs-community/ufs-weather-model/pull/2730 which requires a change of cutoff level 

It also includes updates to use the new spack-stack and porting to URSA.   When spack-stack was updated, GOCART was updated and it includes the following updates (thanks to @bbakernoaa) and are described here: 

This pull request introduces several updates to configuration files for the UFS GOCART model, focusing on adding new parameters, improving wet removal schemes, and updating external data sources. Below is a summary of the most important changes, grouped by theme.

### Wet Removal Scheme Enhancements:
* Added `wet_removal_scheme: ufs` along with rainout efficiency parameters (`fwet_ice`, `fwet_snow`, `fwet_rain`) to multiple configuration files, including `CA2G_instance_CA.bc.rc`, `CA2G_instance_CA.br.rc`, `CA2G_instance_CA.oc.rc`, `DU2G_instance_DU.rc`, and `SS2G_instance_SS.rc`. These parameters define the efficiency of wet removal processes for different precipitation types. [[1]](diffhunk://#diff-868866b7abe13249cb35169ab2fc4ae9bcb56fc0777518d886778451dc08e214R19-R32) [[2]](diffhunk://#diff-9b5c009932e4d4d5ec402ac2f5815d461dbe7d91cb9a10a62a7e0df409ce5367R24-R35) [[3]](diffhunk://#diff-ec2da60e5d01cc93177afdad34eb646cff906daa081a1142a40a65952e7cfef3R29-R42) [[4]](diffhunk://#diff-d9370b3a0a2be8cc6de05c1b6b6c97913e3c15d7e52e3dbb1e3498e8f73d6e99R25-R35) [[5]](diffhunk://#diff-96bb0170283b2fb4ba4eef7133c12ce37c215e4b007cf6e67022c1e063c96c13R43-R50)

### New Parameters for Physical and Chemical Processes:
* Introduced `time_days_hydrophobic_to_hydrophilic` (conversion rate of hydrophobic to hydrophilic aerosols) and `time_days_chemical_destruction` (rate of chemical destruction of carbon species) in `CA2G_instance_CA.bc.rc`, `CA2G_instance_CA.br.rc`, and `CA2G_instance_CA.oc.rc`. These parameters enhance the representation of aerosol lifecycle processes. [[1]](diffhunk://#diff-868866b7abe13249cb35169ab2fc4ae9bcb56fc0777518d886778451dc08e214R19-R32) [[2]](diffhunk://#diff-9b5c009932e4d4d5ec402ac2f5815d461dbe7d91cb9a10a62a7e0df409ce5367R24-R35) [[3]](diffhunk://#diff-ec2da60e5d01cc93177afdad34eb646cff906daa081a1142a40a65952e7cfef3R29-R42)
* Added `soil_clay_factor` in `DU2G_instance_DU.rc` to scale clay fraction in dust emissions.

### Updates to External Data Sources:
* Updated external data paths in `ExtData.other` for variables such as `DU_RDRAG`, `DU_GVF`, and `DU_LAI`, pointing to new datasets for albedo drag, green vegetation fraction, and leaf area index.
* Separated volcanic emissions into `volcano_srcfilen_explosive` and `volcano_srcfilen_degassing` in `SU2G_instance_SU.rc`, improving the handling of volcanic SO₂ sources.

### Dust Emission Scheme Adjustments:
* Modified the `alpha` parameter in the Fengsha dust emission scheme from `0.1` to `0.16` and added `drag_partition_option: 2` in `DU2G_instance_DU.rc`, refining the dust emission calculations.<!--
 


# Type of change
- [ ] Bug fix (fixes something broken)
- [ ] New feature (adds functionality)
- [x] Maintenance (code refactor, clean-up, new CI test, etc.)  Routine model update

# Change characteristics
<!-- Choose YES or NO from each of the following and delete the other -->
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? YES 

- [ ] We will need an update to GDAS App due to MOM6 updates.  Not sure of the GDAS App PR yet, but SOCA PR is https://github.com/JCSDA-internal/soca/pull/1177
- [x] Fix file update: https://github.com/NOAA-EMC/global-workflow/issues/3864

# How has this been tested?
Full CI on Gaea C6 (will post log in comment) 

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have documented my code, including function, input, and output descriptions
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] This change is covered by an existing CI test or a new one has been added
- [x] Any new scripts have been added to the .github/CODEOWNERS file with owners
- [x] I have made corresponding changes to the system documentation if necessary

JessicaMeixner-NOAA committed:
update to new ufs model

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into updatemodel20250625

JessicaMeixner-NOAA commented:
Gaea tests can be found here: /gpfs/f6/ira-sti/world-shared/Jessica.Meixner/updatemodel20250625/RUNTESTS/EXPDIR

```
******** C48_ATM ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Jun 26 2025 13:10:25    Jun 26 2025 18:25:07

******** C48mx500_3DVarAOWCDA ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    Jun 26 2025 13:10:25    Jun 26 2025 13:30:06
202103250000        Done    Jun 26 2025 13:10:25    Jun 26 2025 22:55:11

******** C48mx500_hybAOWCDA ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    Jun 26 2025 13:10:25    Jun 26 2025 13:35:09
202103250000        Done    Jun 26 2025 13:10:25    Jun 26 2025 21:10:19

******** C48_S2SW ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Jun 26 2025 13:10:26    Jun 26 2025 18:25:08

******** C48_S2SWA_gefs ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Jun 26 2025 13:10:25    Jun 27 2025 11:25:11

******** C96_atm3DVar ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    Jun 26 2025 13:10:25    Jun 26 2025 13:35:10
202112210000        Done    Jun 26 2025 13:10:25    Jun 27 2025 11:25:11
202112210600        Done    Jun 26 2025 13:10:25    Jun 27 2025 10:05:09

******** C96C48_hybatmDA ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    Jun 26 2025 13:10:25    Jun 26 2025 13:35:09
202112210000        Done    Jun 26 2025 13:10:25    Jun 27 2025 11:35:06
202112210600        Done    Jun 26 2025 13:10:25    Jun 27 2025 10:05:08

******** C96C48_hybatmsnowDA ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Jun 26 2025 13:10:25    Jun 26 2025 13:30:06
202112201800        Done    Jun 26 2025 13:10:25    Jun 27 2025 11:25:11
202112210000        Done    Jun 26 2025 13:10:25    Jun 27 2025 10:00:07

******** C96C48mx500_S2SW_cyc_gfs ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Jun 26 2025 13:10:25    Jun 26 2025 13:35:09
202112201800        Done    Jun 26 2025 13:10:25    Jun 27 2025 11:35:06
202112210000        Done    Jun 26 2025 13:10:25    Jun 27 2025 12:05:19
202112211800        Done    Jun 26 2025 13:40:05    Jun 27 2025 11:55:11

******** C96_gcafs_cycled ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Jun 26 2025 13:10:25    Jun 26 2025 14:25:05
202112201800        Done    Jun 26 2025 13:10:25    Jun 26 2025 21:25:06
202112210000        Done    Jun 26 2025 13:10:25    Jun 26 2025 17:55:05

******** C96mx100_S2S ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
199405010000        Done    Jun 26 2025 13:10:25    Jun 27 2025 09:45:10


```

JessicaMeixner-NOAA committed:
rt_cutoff should only be changed for C768 and higher resolutions

JessicaMeixner-NOAA commented:
There has been a minor update to the physics settings to appropriately match what is here: https://docs.google.com/document/d/1LOJ0Mip2ZXOHKQxIA_Pkv6GlTXUnOHQbFQl2Psuf2TQ/edit?tab=t.0 which was originally incorrectly updated for lower resolutions.   Thanks to @yangfanglin for spotting this. 


@XiaqiongZhou-NOAA - this PR is ready for your review.

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into updatemodel20250625

JessicaMeixner-NOAA committed:
update model to latest

JessicaMeixner-NOAA commented:
2 more UFS PRs have made their way through the PR queue.

XiaqiongZhou-NOAA reviewed (commented):
It looks good to me.

bingfu-NOAA commented:
> There has been a minor update to the physics settings to appropriately match what is here: https://docs.google.com/document/d/1LOJ0Mip2ZXOHKQxIA_Pkv6GlTXUnOHQbFQl2Psuf2TQ/edit?tab=t.0 which was originally incorrectly updated for lower resolutions. Thanks to @yangfanglin for spotting this.
> 
> @XiaqiongZhou-NOAA - this PR is ready for your review.

Recent GEFSv13 reforecast evaluation suggest we might need this update (rf_cutoff =300) too. @XiaqiongZhou-NOAA might provide more input shortly.

yangfanglin commented:
@XiaqiongZhou-NOAA @bingfu-NOAA we probably should set rf_cutoff=300.0 for all resolutions

JessicaMeixner-NOAA commented:
> @XiaqiongZhou-NOAA @bingfu-NOAA we probably should set rf_cutoff=300.0 for all resolutions

Okay - so you want rt_cutoff=300 for all resolutions and also set for GEFS?  What about SFS?

XiaqiongZhou-NOAA commented:
SFS can also use rf_cutoff=300

JessicaMeixner-NOAA committed:
update rt_cutoff for all resolutions all applications

JessicaMeixner-NOAA commented:
I've updated rt_cuttoff and requested reviews from you all to confirm that I have updated as it should be.

yangfanglin reviewed (approved):
looks good

JessicaMeixner-NOAA commented:
NOTE: this PR since we went to top of ufs now needs and update to gdas app see:  NOAA-EMC/GDASApp#1778

bbakernoaa committed:
adding gocart changes for head of develop

JessicaMeixner-NOAA committed:
Merge pull request #16 from bbakernoaa/feature/gocart_settings

adding gocart changes for head of develop

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into updatemodel20250625

JessicaMeixner-NOAA committed:
update ufs to top of develop

JessicaMeixner-NOAA commented:
I have run tests on Gaea, and everything passed except for 2 tests which had the same issue of input files not being accessible.  Note, this has manually updated GDASApp updates for soca.   @DavidNew-NOAA is working on getting soca updated officially.

JessicaMeixner-NOAA committed:
Change directory paths now that data is staged

JessicaMeixner-NOAA commented:
Running a test on Gaea C6 w/local changes updating GDASApp to it's develop branch + updating SOCA to JCSDA develop + updating mom6 & icepak links I get the following: 

```
******** C48_ATM_um01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Jul 11 2025 14:15:10    Jul 11 2025 15:10:08

******** C48mx500_3DVarAOWCDA_um01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    Jul 11 2025 14:15:11    Jul 11 2025 14:30:16
202103250000        Done    Jul 11 2025 14:15:11    Jul 11 2025 16:00:16

******** C48mx500_hybAOWCDA_um01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    Jul 11 2025 14:15:10    Jul 11 2025 14:30:30
202103250000        Done    Jul 11 2025 14:15:10    Jul 11 2025 15:35:14

******** C48_S2SWA_gefs_um01 ********
202103231200    gefs_fcst_mem000_seg1                   209570882                DEAD                  15         2          79.0
202103231200    gefs_fcst_mem001_seg1                   209571099                DEAD                  15         2          84.0
202103231200    gefs_fcst_mem002_seg1                   209571100                DEAD                  15         2          83.0

******** C48_S2SW_um01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Jul 11 2025 14:15:09    Jul 11 2025 15:30:10

******** C96_atm3DVar_um01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    Jul 11 2025 14:15:09    Jul 11 2025 14:30:13
202112210000        Done    Jul 11 2025 14:15:09    Jul 11 2025 16:20:17
202112210600        Done    Jul 11 2025 14:15:09    Jul 11 2025 16:25:17

******** C96C48_hybatmDA_um01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    Jul 11 2025 14:15:09    Jul 11 2025 14:30:16
202112210000        Done    Jul 11 2025 14:15:09    Jul 11 2025 16:15:17
202112210600        Done    Jul 11 2025 14:15:09    Jul 11 2025 16:30:11

******** C96C48_hybatmsnowDA_um01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Jul 11 2025 14:15:10    Jul 11 2025 14:30:13
202112201800        Done    Jul 11 2025 14:15:10    Jul 11 2025 16:15:35
202112210000        Done    Jul 11 2025 14:15:10    Jul 11 2025 16:20:18

******** C96C48mx500_S2SW_cyc_gfs_um01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Jul 11 2025 14:15:12    Jul 11 2025 14:30:13
202112201800        Done    Jul 11 2025 14:15:12    Jul 11 2025 16:25:17
202112210000        Done    Jul 11 2025 14:15:12    Jul 11 2025 17:46:50
202112211800        Done    Jul 11 2025 14:35:12    Jul 11 2025 17:25:05

******** C96_gcafs_cycled_um01 ********
202112201800         gcafs_fcst_seg0                   209571757                DEAD                  15         2          67.0
202112210000         gcdas_arch_vrfy                   209572104              QUEUED                   -         0           0.0

******** C96mx100_S2S_um01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
199405010000        Done    Jul 11 2025 14:15:10    Jul 11 2025 15:35:14
```


The cases that fail are all using GOCART.  The failures are some flavor of: 

`NaN in input field of mpp_reproducing_sum(_2d), this indicates numerical instability
`

@bbakernoaa do you think this is GOCART related?  Is there an easy way to move forward temporarily if code changes to gocart are needed? 

In the meantime, I'll create a branch from an earlier commit that updates ufs to the commit before gocart was updated + spack-stack & ursa updates in case we need/want to move forward temporarily with that.  We likely need to sync some updates of UFS & GDAS app after some MOM6 changes.   

FYI @RussTreadon-NOAA @DavidNew-NOAA @DavidHuber-NOAA @CoryMartin-NOAA

RussTreadon-NOAA commented:
**GaeaC6 test**

Install `JessicaMeixner-NOAA:updatemodel20250625` at [d7c3747](https://github.com/NOAA-EMC/GDASApp/commit/d7c37479793f1359725eda90078948f0f855ebea) on Gaea C6.  Clone GDASApp [`feature/stable-nightly`](https://github.com/NOAA-EMC/GDASApp/tree/feature/stable-nightly) at [`d7c3747`](https://github.com/NOAA-EMC/GDASApp/commit/d7c37479793f1359725eda90078948f0f855ebea) into `sorc/gdas.cd`.

Job 20211220 18Z gcafs_fcst_seg0 aborts with the `NaN in input field of mpp_reproducing_sum(_2d), this indicates numerical instability` message reported by @JessicaMeixner-NOAA.  Interestingly, the 12Z and 18Z gcdas_fcst_seg0 run to completion without error. 

The successful gcdas_fcst log files print values for pm25 and pm10 after the seas* listings
```
  0:  seas1  0.3588860     -1.8393183E-03  1.5446769E-02
  0:  seas2   11.75535     -3.8298141E-04  0.5020168
  0:  seas3   126.1245      -1.099101       5.207033
  0:  seas4   558.3942      -22.18718       21.39775
  0:  seas5   410252.9      0.0000000E+00   70.89510
  0:  pm25   3901.626      6.6633926E-23   12.21849
  0:  pm10   20569.14      6.6633926E-23   49.49861
  0:  U  max =    133.9300      min =   -148.6235
  0:  V  max =    149.9780      min =   -153.4004
  0:  W   0.4628544     -0.2958416     -5.5361335E-04
  0:
  0:  mp_top=           1 pfull=  1.2781459E-02
```
whereas the failed gcafs_fcst_seg0 aborts before printing the pm25 statistics
```
  0:  seas1  0.2960966      0.0000000E+00  1.5077543E-02
  0:  seas2   9.699719      0.0000000E+00  0.4938292
  0:  seas3   104.2281      0.0000000E+00   5.396657
  0:  seas4   464.8660      0.0000000E+00   27.22557
  2:
  2: FATAL from PE     2: NaN in input field of mpp_reproducing_sum(_2d), this indicates numerical instability
  2:
```

Is this a clue?   Why does the gcdas_fcst run while the gcafs_fcst_seg0 aborts?  I am not familiar with the gcdas and gcafs run of `gcafs_model.x`. How do these runs differ in their input and configuration?

JessicaMeixner-NOAA commented:
Tagging @bbakernoaa for above error messages.

RussTreadon-NOAA commented:
@JessicaMeixner-NOAA :  Have you executed `gfs_model.x` built from JessicaMeixner-NOAA:updatemodel20250625 at [`bcf20f9`](https://github.com/JessicaMeixner-NOAA/global-workflow/commit/bcf20f90c37c5c28eb42f209bbda7601d95ba25a) on Hera?  

I did and encounter the following error messages when `gfs_model.x` is executed
```
+ cpreq[3]cp /scratch1/NCEPDEV/da/Russ.Treadon/CI/GDASApp/stable/20250714/global-workflow/exec/gfs_model.x /scratch1/NCEPDEV/stmp2/Russ.Treadon/RUNDIRS/C48mx500_3DVarAOWCDA/gdas.2021032418/gdasfcst.2021032418/fcst.3699126/
+ cpreq[4]'[' 0 -ne 0 ']'
+ exglobal_forecast.sh[179]srun -l --export=ALL --hint=nomultithread -n 80 /scratch1/NCEPDEV/stmp2/Russ.Treadon/RUNDIRS/C48mx500_3DVarAOWCDA/gdas.2021032418/gdasfcst.2021032418/fcst.3699126/gfs_model.x
10: /scratch1/NCEPDEV/stmp2/Russ.Treadon/RUNDIRS/C48mx500_3DVarAOWCDA/gdas.2021032418/gdasfcst.2021032418/fcst.3699126/gfs_model.x: /apps/gnu/gcc-9.2.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /scratch1/NCEPDEV/stmp2/Russ.Treadon/RUNDIRS/C48mx500_3DVarAOWCDA/gdas.2021032418/gdasfcst.2021032418/fcst.3699126/gfs_model.x)
10: /scratch1/NCEPDEV/stmp2/Russ.Treadon/RUNDIRS/C48mx500_3DVarAOWCDA/gdas.2021032418/gdasfcst.2021032418/fcst.3699126/gfs_model.x: /apps/gnu/gcc-9.2.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.32' not found (required by /scratch1/NCEPDEV/stmp2/Russ.Treadon/RUNDIRS/C48mx500_3DVarAOWCDA/gdas.2021032418/gdasfcst.2021032418/fcst.3699126/gfs_model.x)
```

Should we expect the updated UFS weather model to build and run on Hera or is Hera no longer supported given the transition to Ursa?

JessicaMeixner-NOAA commented:
I have not run on hera, I will try this now as well as trying some updates to turn off radiative feedback from aerosols as a temporary workaround to the failed tests.  I've talked to @bbakernoaa  offline and he's working on some fixes for what should help the model instabilities but those will require an update to GOCART and we need to figure out a way to move forward in the meantime.  

I would think hera would work, but I also know the g-w team didn't plan on continuing support for hera and if hera /scratch1&2 are read only starting tomorrow- my guess is that's the day hera support goes away but @aerorahul @DavidHuber-NOAA will have to confirm.

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into updatemodel20250625

RussTreadon-NOAA commented:
Good reminder about Hera `scratch1/2` becoming read only tomorrow.  There's little point in using Hera for development work.

bbakernoaa commented:
> I have not run on hera, I will try this now as well as trying some updates to turn off radiative feedback from aerosols as a temporary workaround to the failed tests. I've talked to @bbakernoaa offline and he's working on some fixes for what should help the model instabilities but those will require an update to GOCART and we need to figure out a way to move forward in the meantime.
> 
> I would think hera would work, but I also know the g-w team didn't plan on continuing support for hera and if hera /scratch1&2 are read only starting tomorrow- my guess is that's the day hera support goes away but @aerorahul @DavidHuber-NOAA will have to confirm.

The issue is described more at https://github.com/ufs-community/ufs-weather-model/issues/2809. There is also a PR at NASA's authoritative now https://github.com/GEOS-ESM/GOCART/pull/345.  Trying to rush this through ASAP.

JessicaMeixner-NOAA commented:
In the meantime I have two paths forward I'm looking into: 

* Reverting back part of the UFS update to before the GOCART update  + using GDAS app feature/nightly-stable branch: https://github.com/JessicaMeixner-NOAA/global-workflow/tree/updatemodel_pregocart 
* Using this PR but looking to turn off the feedback from GOCART to FV3 to see if that gets the numerical stability resolved while the robust fix from @bbakernoaa  is making it's way through ufs-weather-model

JessicaMeixner-NOAA commented:
@bbakernoaa  - can you test your gocart updates w/the g-w ? Or do I or others need to make those tests to make sure your update resolves the crashes we are seeing?

bbakernoaa commented:
> In the meantime I have two paths forward I'm looking into:
> 
> * Reverting back part of the UFS update to before the GOCART update  + using GDAS app feature/nightly-stable branch: https://github.com/JessicaMeixner-NOAA/global-workflow/tree/updatemodel_pregocart
> * Using this PR but looking to turn off the feedback from GOCART to FV3 to see if that gets the numerical stability resolved while the robust fix from @bbakernoaa  is making it's way through ufs-weather-model

@JessicaMeixner-NOAA I'll submit the CI tests with the GOCART fix and post here.

RussTreadon-NOAA commented:
**Ursa test**

Attempt to build `JessicaMeixner-NOAA:updatemodel20250625` at [`e4607f7`](https://github.com/JessicaMeixner-NOAA/global-workflow/commit/e4607f77c7b82a1597f6b75f49b365392492f9cd) with GDASApp branch `feature/stable-nightly` at [`0b798ef`](https://github.com/NOAA-EMC/GDASApp/commit/0b798efd6dc56003eaed451fe80c9fb06f0e5d88).  Multiple g-w components failed to build on Ursa.

```
build_upp.sh failed with status 1!
build_gsi_utils.sh failed with status 1!
build_ufs_utils.sh failed with status 99!
build_gfs_utils.sh failed with status 1!
build_gsi_monitor.sh failed with status 1!
```

Logic to detect the machine does not contain Ursa in the following component specific scripts 
- upp: `sorc/ufs_model.fd/FV3/upp/tests/detect_machine.sh`
- gsi_utils: `sorc/gsi_utils.fd/ush/detect_macihne.sh`
- ufs_utils: `sorc/ufs_utils.fd/sorc/machine-setup.sh`
- gfs_utils: `sorc/gfs_utils.fd/ush/detect_machine.sh`
- gsi_monitor: `sorc/gsi_monitor.fd/ush/detect_machine.sh`

Without successful builds of the above g-w components on Ursa, we can not run g-w CI on Ursa.  GDASApp g-w CI is also unable to fully run on Ursa.

JessicaMeixner-NOAA committed:
update to turn off radiative feedback temporarily while proper fix is
getting worked on

JessicaMeixner-NOAA committed:
point to gdas nightly stable branch to get SOCA updates

JessicaMeixner-NOAA commented:
@RussTreadon-NOAA   - I updated this branch to turn off radaitve feedback from gocart to the atm to see if we can temporarily get around the model failures - I also pointed to the updated gdasapp branch temporarily in this PR.   My tests on gaea are going super slow, so I pushed these updates so I can also test somewhere else with hopefully faster turnaround time. 

In the meantime, I did get successful GEFS, GCAFS tests with https://github.com/JessicaMeixner-NOAA/global-workflow/tree/updatemodel_pregocart -- This updates UFS only partially and does not include the spack-stack updates or Ursa porting.  Ideally I think we would like to go for the temporary update of turning off radative feedback while the GOCART PR goes through.  @bbakernoaa - we still need tests to pass, but in the meantime, let me know if this does not seem like a reasonable temporary workaround.

JessicaMeixner-NOAA commented:
@bbakernoaa - I finally got my tests to run on gaea and wcoss2, here's the report for wcoss2 for this PR as is: 

```
******** C48_ATM_nft01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Jul 14 2025 21:55:27    Jul 14 2025 22:55:39

******** C48mx500_3DVarAOWCDA_nft01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    Jul 14 2025 21:55:15    Jul 14 2025 22:15:43
202103250000        Done    Jul 14 2025 21:55:15    Jul 14 2025 23:41:09

******** C48mx500_hybAOWCDA_nft01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    Jul 14 2025 21:55:14    Jul 14 2025 22:15:36
202103250000        Done    Jul 14 2025 21:55:14    Jul 14 2025 23:15:36

******** C48_S2SWA_gefs_nft01 ********
202103231200    gefs_fcst_mem000_seg1                   210610313                DEAD                 271         2         112.0
202103231200    gefs_fcst_mem001_seg1                   210610314                DEAD                 271         2         108.0
202103231200    gefs_fcst_mem002_seg1                   210610315                DEAD                 271         2         112.0

******** C48_S2SW_extended_nft01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Jul 14 2025 21:55:13    Jul 15 2025 00:20:35
202103231800        Done    Jul 14 2025 21:55:13    Jul 15 2025 00:25:35

******** C48_S2SW_nft01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Jul 14 2025 21:55:09    Jul 14 2025 23:00:55

******** C96_atm3DVar_extended_nft01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    Jul 14 2025 21:55:21    Jul 14 2025 22:15:34
202112210000        Done    Jul 14 2025 21:55:21    Jul 15 2025 02:35:31
202112210600        Done    Jul 14 2025 21:55:21    Jul 15 2025 03:20:42
202112211200        Done    Jul 14 2025 22:20:50    Jul 15 2025 05:30:54
202112211800        Done    Jul 15 2025 02:40:41    Jul 15 2025 09:15:35

******** C96C48_hybatmDA_nft01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    Jul 14 2025 21:55:13    Jul 14 2025 22:15:34
202112210000        Done    Jul 14 2025 21:55:13    Jul 15 2025 00:30:58
202112210600        Done    Jul 14 2025 21:55:13    Jul 15 2025 00:05:33

******** C96C48_hybatmsnowDA_nft01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Jul 14 2025 21:55:10    Jul 14 2025 22:20:57
202112201800        Done    Jul 14 2025 21:55:10    Jul 15 2025 00:30:49
202112210000        Done    Jul 14 2025 21:55:10    Jul 15 2025 00:15:45

******** C96C48mx500_S2SW_cyc_gfs_nft01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Jul 14 2025 21:55:08    Jul 14 2025 22:15:37
202112201800        Done    Jul 14 2025 21:55:08    Jul 15 2025 00:35:35
202112210000        Done    Jul 14 2025 21:55:08    Jul 15 2025 01:35:32
202112211800        Done    Jul 14 2025 22:20:23    Jul 15 2025 01:30:47

******** C96C48_ufs_hybatmDA_nft01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202402231800        Done    Jul 14 2025 21:55:10    Jul 14 2025 22:15:31
202402240000        Done    Jul 14 2025 21:55:10    Jul 15 2025 00:50:42
202402240600        Done    Jul 14 2025 21:55:10    Jul 15 2025 00:50:42

******** C96_gcafs_cycled_nft01 ********
202112201800         gcafs_fcst_seg0                   210617873                DEAD                 271         2          96.0

******** C96_gcafs_cycled_noDA_nft01 ********
202112201800         gcafs_fcst_seg0                   210616463                DEAD                 271         2         117.0

******** C96mx100_S2S_nft01 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
199405010000        Done    Jul 14 2025 21:55:11    Jul 14 2025 23:35:35
```

We are still getting forecast failures with GOCART.  

I'm going to open a PR that updates UFS to the commit prior to GOCART getting updated, but that means no one can move to the new spack-stack or have Ursa in global-workflow until there is a workaround.  Thoughts on other temporary workarounds?  Were other things needed to be changed beyond IAER?   Do you know how long it'll take before a GOCART PR with a fix is ready?  Did your tests with the gocart update succeed?

JessicaMeixner-NOAA commented:
@DavidHuber-NOAA @RussTreadon-NOAA @DavidNew-NOAA @bbakernoaa  - While issues with model related issues on this PR are being worked on, I made a PR here: https://github.com/NOAA-EMC/global-workflow/pull/3872 that should pass all regression tests, allow GDASApp to get updated, etc.   I'm not sure how long it'll take to resolve issues with this PR so it might be a good idea to let the other PR move forward since we had a chicken-and-egg problem with the GDASApp, which would at least let both of those move forward.

RussTreadon-NOAA committed:
Update Hera and GaeaC6 configuration for select JEDI jobs (#3868)

This modifies the configuration for the following jobs on the
following machines
- Hera:  aeroanlvar
- Gaea C6:  atmanlvarm, atmensanlobs, atmensanlsol

For each job on each machine the change reduces the number of tasks run
per node. it has been observed that some executables require more memory
after moving to spack-stack/1.9.2 This is the case for the above jobs on
the indicated machines.

Resolves #3862

tsga committed:
Add GSI Soil DA Increments to Archive  (#3857)

# Description
This adds the names of soil DA increment files from GSI EnKF to the
"archive_tars" lists. Currently soil forecast and analysis files are
being archived, but not soil increments.

Resolves #3858

DavidHuber-NOAA commented:
Adding the blocked label to this PR.  @JessicaMeixner-NOAA please remove the label when this PR is ready for re-review.

jiaruidong2017 committed:
Add snow DA related fields and stats into the gfsa tarball (#3842)

This adds the snow DA related files into `gfsa.tar` for HPSS archive.
The snow DA related files include snow analysis, snow increment,
diagnostic files, snow DA yaml file and snow_iodastat, which is
consistent with the gdas run.

Resolves #3841

DavidHuber-NOAA committed:
Ignore 'unused' variables in config.resources* files (#3879)

This adds `#shellcheck disable=SC2034` to all `config.resources*` files
to prevent erroneously detecting 'unused' variables.

This was was performed by copilot in
https://github.com/DavidHuber-NOAA/global-workflow/pull/28.

Resolves #3876

DavidHuber-NOAA committed:
HOTFIX: Update the CODEOWNERS file (#3880)

This updates the CODEOWNERS file so @aerorahul and @DavidHuber-NOAA are
requested as reviewers on all PRs. Previously, these lines were treated
as comments.

JessicaMeixner-NOAA committed:
update gdas hash

JessicaMeixner-NOAA committed:
undo change to turn off feedback as that did not fix model crash
update gdas app hash to dev branch

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into updatemodel20250625

JessicaMeixner-NOAA committed:
fix whitespace changes

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into updatemodel20250625

JessicaMeixner-NOAA commented:
Since a lot has gone on in this PR, giving a quick summary. 

Part of this PR was merged in https://github.com/NOAA-EMC/global-workflow/pull/3872 which included all of the physics updates and UFS updates right before the move to spack-stack. 

To update UFS further in develop, which includes the spack-stack 1.9.2 updates and Ursa porting, we will need to update UFS further to include an additional update to GOCART (see issue here: https://github.com/ufs-community/ufs-weather-model/issues/2809).  In the meantime, for anytest that does not use aersols, you sould be able to use this branch successfully (ie you could run on Ursa).

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into updatemodel20250625

JessicaMeixner-NOAA committed:
update config for model update

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into updatemodel20250625

JessicaMeixner-NOAA commented:
Testing with UFS PR https://github.com/ufs-community/ufs-weather-model/pull/2819: 


******** C48_ATM_gup02 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Jul 21 2025 21:15:06    Jul 21 2025 23:05:06

******** C48mx500_3DVarAOWCDA_gup02 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    Jul 21 2025 21:15:06    Jul 21 2025 22:15:22
202103250000        Done    Jul 21 2025 21:15:06    Jul 21 2025 23:40:13

******** C48mx500_hybAOWCDA_gup02 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    Jul 21 2025 21:15:06    Jul 21 2025 22:25:08
202103250000        Done    Jul 21 2025 21:15:06    Jul 21 2025 23:30:22

******** C48_S2SWA_gefs_gup02 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Jul 21 2025 21:15:07    Jul 21 2025 23:15:10

******** C48_S2SW_gup02 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Jul 21 2025 21:15:06    Jul 21 2025 22:55:04

******** C96_atm3DVar_gup02 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    Jul 21 2025 21:15:07    Jul 21 2025 22:25:09
202112210000        Done    Jul 21 2025 21:15:07    Jul 22 2025 00:20:05
202112210600        Done    Jul 21 2025 21:15:07    Jul 22 2025 00:20:05

******** C96C48_hybatmDA_gup02 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    Jul 21 2025 21:15:06    Jul 21 2025 22:15:22
202112210000        Done    Jul 21 2025 21:15:06    Jul 22 2025 00:20:05
202112210600        Done    Jul 21 2025 21:15:06    Jul 22 2025 00:30:07

******** C96C48_hybatmsnowDA_gup02 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Jul 21 2025 21:15:06    Jul 21 2025 22:25:08
202112201800        Done    Jul 21 2025 21:15:06    Jul 22 2025 00:20:05
202112210000        Done    Jul 21 2025 21:15:06    Jul 22 2025 00:15:30

******** C96C48mx500_S2SW_cyc_gfs_gup02 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Jul 21 2025 21:15:06    Jul 21 2025 22:25:09
202112201800        Done    Jul 21 2025 21:15:06    Jul 22 2025 00:30:07
202112210000        Done    Jul 21 2025 21:15:06    Jul 22 2025 01:35:26
202112211800        Done    Jul 21 2025 22:30:07    Jul 22 2025 01:25:12

******** C96_gcafs_cycled_gup02 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Jul 21 2025 21:15:06    Jul 21 2025 22:15:22
202112201800        Done    Jul 21 2025 21:15:06    Jul 21 2025 23:55:05
202112210000        Done    Jul 21 2025 21:15:06    Jul 21 2025 23:35:20

******** C96_gcafs_cycled_noDA_gup02 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Jul 21 2025 21:15:07    Jul 21 2025 22:15:22
202112201800        Done    Jul 21 2025 21:15:07    Jul 21 2025 23:40:13
202112210000        Done    Jul 21 2025 21:15:07    Jul 21 2025 23:10:17

******** C96mx100_S2S_gup02 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
199405010000        Done    Jul 21 2025 21:15:06    Jul 21 2025 23:20:29

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into updatemodel20250625

JessicaMeixner-NOAA committed:
updates for model

JessicaMeixner-NOAA commented:
PR  https://github.com/ufs-community/ufs-weather-model/pull/2819 that is blocking this PR in UFS is currently in testing.  I have updated this branch locally to point to that UFS branch and am running on Gaea C6: 
./generate_workflows.sh -b -D -G -E -S -C -A ira-sti -c -t um0731 /gpfs/f6/ira-sti/world-shared/Jessica.Meixner/updatemodel20250731/um0731

So that this PR can hopefully move forward with review and testing as soon as that PR is merged.

JessicaMeixner-NOAA commented:
Update on tests from Gaea.  As soon as the UFS PR is merged, I will update the hash and remove the blocked label here.  

******** C48_ATM_um0731 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Jul 31 2025 13:55:19    Jul 31 2025 17:00:07

******** C48mx500_3DVarAOWCDA_um0731 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    Jul 31 2025 13:55:19    Jul 31 2025 14:12:01
202103250000        Done    Jul 31 2025 13:55:19    Jul 31 2025 20:10:17

******** C48mx500_hybAOWCDA_um0731 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    Jul 31 2025 13:55:19    Jul 31 2025 14:12:01
202103250000        Done    Jul 31 2025 13:55:19    Jul 31 2025 18:45:06

******** C48_S2SWA_gefs_um0731 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Jul 31 2025 13:55:20    Jul 31 2025 19:00:09

******** C48_S2SW_um0731 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Jul 31 2025 13:55:20    Jul 31 2025 17:35:10

******** C96_atm3DVar_um0731 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    Jul 31 2025 13:55:21    Jul 31 2025 14:12:01
202112210000        Done    Jul 31 2025 13:55:21    Aug 01 2025 02:20:22
202112210600        Done    Jul 31 2025 13:55:21    Jul 31 2025 20:00:20

******** C96C48_hybatmDA_um0731 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    Jul 31 2025 13:55:20    Jul 31 2025 14:12:01
202112210000        Done    Jul 31 2025 13:55:20    Aug 01 2025 02:35:23
202112210600        Done    Jul 31 2025 13:55:20    Jul 31 2025 20:30:20

******** C96C48_hybatmsnowDA_um0731 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Jul 31 2025 13:55:19    Jul 31 2025 14:12:03
202112201800        Done    Jul 31 2025 13:55:19    Aug 01 2025 02:40:06
202112210000        Done    Jul 31 2025 13:55:19    Jul 31 2025 20:20:11

******** C96C48mx500_S2SW_cyc_gfs_um0731 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Jul 31 2025 13:55:20    Jul 31 2025 14:12:01
202112201800        Done    Jul 31 2025 13:55:20    Aug 01 2025 02:35:22
202112210000        Done    Jul 31 2025 13:55:20    Aug 01 2025 02:55:23
202112211800        Done    Jul 31 2025 14:15:19    Aug 01 2025 02:10:08

******** C96_gcafs_cycled_noDA_um0731 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Jul 31 2025 13:55:20    Jul 31 2025 14:15:19
202112201800        Done    Jul 31 2025 13:55:20    Jul 31 2025 18:55:19
202112210000        Done    Jul 31 2025 13:55:20    Jul 31 2025 18:00:07

******** C96_gcafs_cycled_um0731 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Jul 31 2025 13:55:21    Jul 31 2025 14:15:19
202112201800        Done    Jul 31 2025 13:55:21    Jul 31 2025 19:20:13
202112210000        Done    Jul 31 2025 13:55:21    Jul 31 2025 19:05:07

******** C96mx100_S2S_um0731 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
199405010000        Done    Jul 31 2025 13:55:19    Jul 31 2025 18:50:10

JessicaMeixner-NOAA committed:
update model

JessicaMeixner-NOAA commented:
This PR is officially unblocked!!!  (Thank you @bbakernoaa !!!) 

@DavidHuber-NOAA  - This PR is ready for it's next steps.

DavidHuber-NOAA reviewed (dismissed):
Changes look good.

DavidHuber-NOAA commented:
Launching CI on all Hera, Hercules, and WCOSS2.  C6 testing by @JessicaMeixner-NOAA is sufficient for that platform.

@TerrenceMcGuinness-NOAA Side note, I will be launching Hera CI via generate_workflows.sh.

DavidHuber-NOAA commented:
Note that I won't be setting any Hera labels to avoid triggering anything Jenkins-related.

JessicaMeixner-NOAA commented:
All the errors on hercules that I've looked at so far have been srun errors.  Installing a generate_workflows to test out from my account so I can try re-running/debugging if necessary.

DavidHuber-NOAA commented:
It looks like Hera may need a module file update.  All of the forecasts failed at launch time with GLIBCXX errors.  For C48_ATM

```
exglobal_forecast.sh[179]srun -l --export=ALL --hint=nomultithread -n 40 /scratch3/NCEPDEV/stmp/David.Huber/RUNDIRS/C48_ATM_3840/gfs.2021032312/gfsfcst.2021032312/fcst.4013214/gfs_model.x
12: /scratch3/NCEPDEV/stmp/David.Huber/RUNDIRS/C48_ATM_3840/gfs.2021032312/gfsfcst.2021032312/fcst.4013214/gfs_model.x: /apps/gnu/gcc-9.2.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /scratch3/NCEPDEV/stmp/David.Huber/RUNDIRS/C48_ATM_3840/gfs.2021032312/gfsfcst.2021032312/fcst.4013214/gfs_model.x)
```

JessicaMeixner-NOAA commented:
I'm wondering if hercules needs a module and/or environment update or if srun was just having errors this weekend?  My re-run had the exact same problems the CI tests had. 

Module update seems odd because don't we use the ufs-weather-model modules in g-w?  And those have been running with what's there. I wonder if there are hera environment variables that need updated versus modules? I'll check out the ufs commit history.

DavidHuber-NOAA commented:
On WCOSS2, the UPP has changed its module names.  upp.sh should be updated with the following patch:

```diff
diff --git a/dev/jobs/upp.sh b/dev/jobs/upp.sh
index 4d070e9c..50b2825e 100755
--- a/dev/jobs/upp.sh
+++ b/dev/jobs/upp.sh
@@ -18,7 +18,7 @@ if [[ "${MACHINE_ID}" == "wcoss2" ]]; then
   set +x
   source "${HOMEgfs}/ush/module-setup.sh"
   module use "${HOMEgfs}/sorc/ufs_model.fd/FV3/upp/modulefiles"
-  module load "${MACHINE_ID}"
+  module load "${MACHINE_ID}_intel"
   module load prod_util
   module load cray-pals
   module load cfp
```

DavidHuber-NOAA commented:
> I'm wondering if hercules needs a module and/or environment update or if srun was just having errors this weekend? My re-run had the exact same problems the CI tests had.
> 
> Module update seems odd because don't we use the ufs-weather-model modules in g-w? And those have been running with what's there. I wonder if there are hera environment variables that need updated versus modules? I'll check out the ufs commit history.

No, we moved to loading the workflow modules for the R&D machines when we updated to spack-stack 1.5.1, but it looks like we need to temporarily switch back to using the UFS modules since the model has upgraded to 1.9.2 and the workflow is still using 1.6.0.

JessicaMeixner-NOAA commented:
How close is your spack-stack PR?  Do we just combine this w/that so you can not have that work around?

JessicaMeixner-NOAA committed:
update upp module

DavidHuber-NOAA commented:
@JessicaMeixner-NOAA I am having issues with the 2nd full cycle gdas_fcst and enkfgdas_efcs jobs for all tests on Hera and MSU.  I would be happy to combine PRs and tackle these issues together, which would also allow me to remove a kludge in `fcst.sh`.

Thoughts @aerorahul?

JessicaMeixner-NOAA commented:
We should at minimum see if combining helps resolve issues for both of us maybe?  The sooner we can get part/all of these updates in the better.  

I'll see if on hercules if i load ufs-modules if any of the forecasts will go through.

DavidHuber-NOAA commented:
@JessicaMeixner-NOAA Sounds good to me.  I'll give that a try on Hera.

JessicaMeixner-NOAA commented:
Okay, I updated the dev/jobs/fcst.sh  job to load the ufs modules, 

I get this module load error 
```
Running "module reset". Resetting modules to system default. The following $MODULEPATH directories have been removed: None
Lmod has detected the following error: Infinite Load Loop detected for module:
"wgrib2/3.7.0" file: "/apps/contrib/modulefiles/wgrib2/3.7.0.lua"
While processing the following module(s):
    Module fullname  Module Filename
    ---------------  ---------------
    wgrib2/3.7.0     /apps/contrib/modulefiles/wgrib2/3.7.0.lua
```

and then I get this error: 

```
+ forecast_postdet.sh[110]/work2/noaa/marine/jmeixner/hercules/gw3840/global-workflow/ush/check_land_input_orography.py --input_dir /work2/noaa/stmp/jmeixner/HERCULES/RUNDIRS/C48_S2SW_up02/gfs.2021032312/gfsfcst.2021032312/fcst.1396406/INPUT --orog_dir /work2/noaa/stmp/jmeixner/HERCULES/RUNDIRS/C48_S2SW_up02/gfs.2021032312/gfsfcst.2021032312/fcst.1396406/INPUT
Traceback (most recent call last):
  File "/work2/noaa/marine/jmeixner/hercules/gw3840/global-workflow/ush/check_land_input_orography.py", line 38, in <module>
    from netCDF4 import Dataset
ModuleNotFoundError: No module named 'netCDF4'
+ JGLOBAL_FORECAST[102]export err=1
+ JGLOBAL_FORECAST[102]err=1
+ JGLOBAL_FORECAST[103][[ 1 -ne 0 ]]
+ JGLOBAL_FORECAST[104]err_exit
+ preamble.sh[109]set +eux
```

This makes me lean towards combining the PRs as the path forward.   That being said - we have run on both Gaea and WCOSS2 with the updated model (not perhaps this exact version but similar).

JessicaMeixner-NOAA commented:
Given that the spack-stack Pr is stuck... I'm going to work on updating this even though many changes will have to be un-done w/the spack-stack PR.

JessicaMeixner-NOAA commented:
@DavidHuber-NOAA - did the WCOSS2 regtests pass?  I'm assuming hera is not going to work until https://github.com/NOAA-EMC/global-workflow/pull/3798 is through?

DavidHuber-NOAA commented:
@JessicaMeixner-NOAA apologies for not posting the results.  Unfortunately, the working space on stmp has been scrubbed, but based on emails from cron, it looks like the UPP jobs failed (both the atmanlupp and goes_upp for the extended tests).  The runtimes were all quite short, so my guess is this was caused by module loads and was probably fixed by https://github.com/NOAA-EMC/global-workflow/pull/3840/commits/6dc014ee251fd9591900b816da445c8ad6885feb.

DavidHuber-NOAA commented:
Let me know when you are ready and I will kick off another suite of tests on Cactus.

JessicaMeixner-NOAA commented:
No worries @DavidHuber-NOAA  - I'll do a few tests on hercules and wcoss2 and let you know when this is ready to be re-tested. 

In the meantime - I'll continue to engage on the spack-stack Pr as well.

JessicaMeixner-NOAA committed:
Merge branch 'NOAA-EMC:develop' into updatemodel20250625

JessicaMeixner-NOAA commented:
On hercules, loading the UWM modules does not work because then we get an error: 

```
+ forecast_postdet.sh[108][[ YES == \Y\E\S ]]
+ forecast_postdet.sh[110]/work2/noaa/marine/jmeixner/hercules/updatemodel/global-workflow/ush/check_land_input_orography.py --input_dir /work2/noaa/stmp/jmeixner/HERCULES/RUNDIRS/t01/gdas.2021122012/gdasfcst.2021122012/fcst.2141940/INPUT --orog_dir /work2/noaa/stmp/jmeixner/HERCULES/RUNDIRS/t01/gdas.2021122012/gdasfcst.2021122012/fcst.2141940/INPUT
Traceback (most recent call last):
  File "/work2/noaa/marine/jmeixner/hercules/updatemodel/global-workflow/ush/check_land_input_orography.py", line 38, in <module>
    from netCDF4 import Dataset
ModuleNotFoundError: No module named 'netCDF4'
+ JGLOBAL_FORECAST[102]export err=1
+ JGLOBAL_FORECAST[102]err=1
+ JGLOBAL_FORECAST[103][[ 1 -ne 0 ]]
+ JGLOBAL_FORECAST[104]err_exit
+ preamble.sh[109]set +eux
```

JessicaMeixner-NOAA commented:
On WCOSS2 - I did get have almost all tests  pass: 

```
******** C48_ATM_upt03 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Aug 08 2025 19:10:26    Aug 08 2025 20:10:51

******** C48mx500_3DVarAOWCDA_upt03 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    Aug 08 2025 19:10:23    Aug 08 2025 19:25:46
202103250000        Done    Aug 08 2025 19:10:23    Aug 08 2025 20:45:46

******** C48mx500_hybAOWCDA_upt03 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    Aug 08 2025 19:10:28    Aug 08 2025 19:25:43
202103250000        Done    Aug 08 2025 19:10:28    Aug 08 2025 20:25:46

******** C48_S2SWA_gefs_upt03 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Aug 08 2025 19:10:20    Aug 08 2025 20:20:55

******** C48_S2SW_extended_upt03 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Aug 08 2025 19:10:31    Aug 08 2025 21:35:39
202103231800        Done    Aug 08 2025 19:10:31    Aug 08 2025 21:40:47

******** C48_S2SW_upt03 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Aug 08 2025 19:10:13    Aug 08 2025 20:10:48

******** C96_atm3DVar_extended_upt03 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    Aug 08 2025 19:10:24    Aug 08 2025 19:30:50
202112210000        Done    Aug 08 2025 19:10:24    Aug 08 2025 23:45:45
202112210600        Done    Aug 08 2025 19:10:24    Aug 09 2025 00:40:54
202112211200        Done    Aug 08 2025 19:35:19    Aug 09 2025 02:50:55
202112211800        Done    Aug 08 2025 23:50:50    Aug 09 2025 06:01:02

******** C96C48_hybatmDA_upt03 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    Aug 08 2025 19:10:12    Aug 08 2025 19:30:57
202112210000        Done    Aug 08 2025 19:10:12    Aug 08 2025 21:35:39
202112210600        Done    Aug 08 2025 19:10:12    Aug 08 2025 21:15:44

******** C96C48_hybatmsnowDA_upt03 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Aug 08 2025 19:10:18    Aug 08 2025 19:30:14
202112201800        Done    Aug 08 2025 19:10:18    Aug 08 2025 21:41:03
202112210000        Done    Aug 08 2025 19:10:18    Aug 08 2025 21:20:18

******** C96C48mx500_S2SW_cyc_gfs_upt03 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Aug 08 2025 19:10:49    Aug 08 2025 19:30:54
202112201800        Done    Aug 08 2025 19:10:49    Aug 08 2025 21:41:02
202112210000        Done    Aug 08 2025 19:10:49    Aug 08 2025 22:50:48
202112211800        Done    Aug 08 2025 19:35:30    Aug 08 2025 22:40:49

******** C96C48_ufs_hybatmDA_upt03 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202402231800        Done    Aug 08 2025 19:10:18    Aug 08 2025 19:35:43
202402240000        Done    Aug 08 2025 19:10:18    Aug 08 2025 22:46:00
202402240600        Done    Aug 08 2025 19:10:18    Aug 08 2025 22:25:57

******** C96_gcafs_cycled_noDA_upt03 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Aug 08 2025 19:10:44    Aug 08 2025 19:31:06
202112201800        Done    Aug 08 2025 19:10:44    Aug 08 2025 21:25:49
202112210000        Done    Aug 08 2025 19:10:44    Aug 08 2025 20:21:01

******** C96_gcafs_cycled_upt03 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Aug 08 2025 19:10:34    Aug 08 2025 19:30:56
202112201800        Done    Aug 08 2025 19:10:34    Aug 08 2025 21:40:47
202112210000        Done    Aug 08 2025 19:10:34    Aug 08 2025 20:55:41

******** C96mx100_S2S_upt03 ********
199405010000    sfs_fcst_mem001_seg1                   213378662                DEAD                 271         2         109.0

```

The SFS test failure is here /lfs/h2ouple/noscrub/jessica.meixner/updateModelDev/upt03/COMROOT/C96mx100_S2S_upt03/logs/1994050100/sfs_fcst_mem001_seg1.log.0

And I'm getting a "permission denied" 


```
nid002545.cactus.wcoss2.ncep.noaa.gov 0:  ==>Note from fv_grid_tools_mod(read_grid): read atmosphere grid from mosaic ver
 sion grid
nid002545.cactus.wcoss2.ncep.noaa.gov 30:
FATAL from PE     0: Permission denied: netcdf_file_open:MOM6_OUTPUT/ocean_geometry.nc

nid002545.cactus.wcoss2.ncep.noaa.gov 30:
FATAL from PE     0: Permission denied: netcdf_file_open:MOM6_OUTPUT/ocean_geometry.nc

nid002545.cactus.wcoss2.ncep.noaa.gov 30: Image              PC                Routine            Line        Source
libpnetcdf.so.4.0  00001504CA10D86A  tracebackqq_          Unknown  Unknown
sfs_model.x        00000000058B45AE  mpp_mod_mp_mpp_er          72  mpp_util_mpi.inc
sfs_model.x        0000000005769AC2  fms_io_utils_mod_         190  fms_io_utils.F90
sfs_model.x        00000000057B5513  netcdf_io_mod_mp_         386  netcdf_io.F90
sfs_model.x        00000000057DA3AE  netcdf_io_mod_mp_         665  netcdf_io.F90
sfs_model.x        000000000576A624  fms_netcdf_domain         388  fms_netcdf_domain_io.F90
sfs_model.x        00000000050EA5FD  mom_io_infra_mp_o         348  MOM_io_infra.F90
sfs_model.x        0000000004A5EA48  mom_io_file_mp_op         947  MOM_io_file.F90
sfs_model.x        0000000004A39615  mom_io_mp_create_         350  MOM_io.F90
sfs_model.x        0000000004B68834  mom_shared_initia        1425  MOM_shared_initialization.F90
sfs_model.x        00000000046E4E98  mom_mp_initialize        3128  MOM.F90
sfs_model.x        00000000046813A6  mom_ocean_model_n         284  mom_ocean_model_nuopc.F90
sfs_model.x        0000000004647D39  mom_cap_mod_mp_in         714  mom_cap.F90
```

This might be an issue for you too @DavidHuber-NOAA  

Also tagging @jiandewang since it's ocean and @XiaqiongZhou-NOAA @NeilBarton-NOAA since it's an SFS test.

DavidHuber-NOAA commented:
I believe this was fully wrapped into #3798.  Can you confirm @JessicaMeixner-NOAA?

JessicaMeixner-NOAA commented:
Confirm - This was merged into https://github.com/NOAA-EMC/global-workflow/pull/3798.
